### DISCUSSION TITLE
Query Explain

### DISCUSSION THREAD

**Discussion by arunkumaralagesan** [trinodb/trino#14539] [Q&A]
Do we have any pretty print mechanisms for the explain output?
It's hard for the users to go over the output and gain insights.

Any solution for a "pretty" rendering in javascript/React would be very much helpful

**Comment by hashhar** [comment]
The Web UI renders the plan as a tree - but note that the "tree" isn't the hard part about the EXPLAIN - it's the various numbers in each node. I'd suggest you to take a look at https://trino.io/blog/2020/07/30/training-query-tuning.html to make more sense of the plans.

There's a graphviz output mode as well which can be rendered to diagrams using `EXPLAIN (FORMAT GRAPHVIZ)`.
